src/Component/Shop/ProductShop/Pagination/index.jsx

```diff
@@ -0,0 +1,90 @@
+import { useSelector, useDispatch } from "react-redux";
+import { useEffect } from "react";
+import {
+  setCurrentPage,
+  getShopProducts,
+  getPagination,
+} from "../../../../store/Slide/ProductsSlide";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faArrowRight, faArrowLeft } from "@fortawesome/free-solid-svg-icons";
+import "./style.scss";
+
+function Pagination({ filter }) {
+  const dispatch = useDispatch();
+  const currentPage = useSelector((state) => state.products.currentPage);
+  const paginations = useSelector((state) => state.products.pagination);
+  const listPage = [];
+  const _page = filter._page;
+
+  useEffect(() => {
+    dispatch(
+      getPagination({
+        ...filter,
+        _page: "",
+        _limit: "",
+      })
+    );
+  }, [filter]);
+
+  if (paginations.list.data) {
+    const allProductsLength = paginations.list.data.length;
+    const listPageLength =
+      allProductsLength % 9
+        ? Math.ceil(allProductsLength / 9)
+        : allProductsLength / 1;
+    for (let i = 1; i <= listPageLength; i++) {
+      listPage.push(i);
+    }
+  }
+
+  const handlePageChange = (num) => {
+    dispatch(getShopProducts({ ...filter, _page: num }));
+    dispatch(setCurrentPage(num));
+  };
+
+  // Xử lý khi có nhiều page
+  const renderPaginationList = () => {
+    const renderBigArr =
+      listPage.length <= 6
+        ? listPage
+        : [_page, _page + 1, _page + 2, "...", listPage.length];
+
+    return renderBigArr.map((item) => {
+      return (
+        <button
+          className={
+            item === currentPage ? "pagination__btn active" : "pagination__btn"
+          }
+          key={item}
+          onClick={() => handlePageChange(item)}
+        >
+          {item}
+        </button>
+      );
+    });
+  };
+
+  return (
+    <section className="pagination d-flex justify-content-center">
+      <div className="pagination_content d-flex align-items-center">
+        <button
+          disabled={currentPage <= 1}
+          className="pagination_action pagination__btn"
```

Comment:
```suggestion
          className="pagination__action pagination__btn"
```
Chỗ thì _ chỗ thì __ là sao nhỉ?
Xem lại về quy tắc đặt tên BEM